lua/weapons/weapon_ace_portablemortar/shared.lua

```diff
@@ -0,0 +1,663 @@
+
+SWEP.ReloadSound = "" --Sound other players hear when you reload - this is NOT your first-person sound
+										--Most models have a built-in first-person reload sound
+
+SWEP.ZoomFOV = 45
+SWEP.HasScope = true --True if the weapon has a sniper-style scope
+
+
+--Recoil (crosshair movement) settings--
+--"Heat" is a number that represents how long you've been firing, affecting how quickly your crosshair moves upwards
+SWEP.HeatReductionRate = 0 --Heat loss per second when not firing
+SWEP.HeatReductionDelay = 0
+SWEP.HeatPerShot = 0 --Heat generated per shot
+SWEP.HeatMax = 0 --Maximum heat - determines max rate at which recoil is applied to eye angles
+				--Also determines point at which random spread is at its highest intensity
+				--HeatMax divided by HeatPerShot gives you how many shots until you reach MaxSpread
+
+SWEP.RecoilSideBias = 0 --How much the recoil is biased to one side proportional to vertical recoil
+						--Positive numbers bias to the right, negative to the left
+
+SWEP.ZoomRecoilBonus = 1 --Reduce recoil by this amount when zoomed or scoped
+SWEP.CrouchRecoilBonus = 1 --Reduce recoil by this amount when crouching
+SWEP.ViewPunchAmount = 8 --Degrees to punch the view upwards each shot - does not actually move crosshair, just a visual effect
+
+
+--Spread (aimcone) settings--
+SWEP.BaseSpread = 4 --First-shot random spread, in degrees
+SWEP.MaxSpread = 4 --Maximum added random spread from heat value, in degrees
+					--If HeatMax is 0 this will be ignored and only BaseSpread will be taken into account (AT4 for example)
+SWEP.MovementSpread = 0 --Increase aimcone to this many degrees when sprinting at full speed
+SWEP.UnscopedSpread = 0 --Spread, in degrees, when unscoped with a scoped weapon
+
+
+--Model settings--
+SWEP.ViewModelFlip = false
+SWEP.ViewModel = "models/weapons/mortar/mortartube.mdl"
+SWEP.WorldModel = "models/weapons/mortar/mortartube.mdl"
+SWEP.HoldType = "normal"
+
+-- Adjust these variables to move the viewmodel's position
+SWEP.IronSightsPos = Vector( 5.2, -5, -4 )
+SWEP.IronSightsAng = Vector( 0, 0, 0 )
+
+SWEP.DeployDelay = 4 --Time before you can fire after deploying the weapon
+SWEP.CSMuzzleFlashes = true
+
+SWEP.FuseDelay = 0
+
+SWEP.CarrySpeedMul			= 0.75
+
+SWEP.LastThink = 0
+SWEP.RangeTick = 0
+SWEP.TargetDistance = 0
+SWEP.CurrentReplenishTicks = 0
+SWEP.RoundType = 0 --0 for He, 1 for smoke
+
+
+DEFINE_BASECLASS("weapon_ace_base")
+
+
+
+if CLIENT then
+	local TubeModel = ClientsideModel("models/weapons/mortar/mortartube.mdl")
+	local BaseModel = ClientsideModel("models/weapons/mortar/mortarbottom.mdl")
+	local BipodModel = ClientsideModel("models/weapons/mortar/mortar_bipod.mdl")
+
+	-- Settings...
+	TubeModel:SetSkin(1)
+	TubeModel:SetNoDraw(true)
+
+	BaseModel:SetSkin(1)
+	BaseModel:SetNoDraw(true)
+
+	BipodModel:SetSkin(1)
+	BipodModel:SetNoDraw(true)
+
+	function SWEP:DrawWorldModel()
+		local _Owner = self:GetOwner()
+
+		if (IsValid(_Owner)) then
```

Comment:
```suggestion
		if IsValid(_Owner) then
```

Unnecessary parenthesis